src/pts/pyspark/intogen.py

```diff
@@ -0,0 +1,159 @@
+    """
+    # Joining cancer driver genes with cohorts:
+    return (
+        genes.join(cohorts, on='COHORT', how='inner')
+        .select(
+            # Adding constant columns:
+            f.lit('somatic_mutation').alias('datatypeId'),
+            f.lit('intogen').alias('datasourceId'),
+            # Extracting gene fields:
+            f.col('SYMBOL').alias('targetFromSourceId'),
+            # Extracting cohort specific annotation:
+            f.col('COHORT').alias('cohortId'),
+            f.col('COHORT_NICK').alias('cohortShortName'),
+            f.col('COHORT_NAME').alias('cohortDescription'),
+            # Splitting methods:
+            f.split(f.col('METHODS'), ',').alias('significantDriverMethods'),
+            # Generating mutated samples column:
+            f.array(
+                f.struct(
+                    # Parse functional consequences:
+                    f.when(f.col('ROLE') == 'Act', 'SO_0002053')  # Gain of function
+                    .when(f.col('ROLE') == 'LoF', 'SO_0002054')  # Loss of function
+                    .otherwise(None)
+                    .alias('functionalConsequenceId'),
+                    # Extract samples:
+                    f.col('SAMPLES').alias('numberSamplesTested'),
+                    f.col('SAMPLES').alias('numberMutatedSamples'),
+                )
+            ).alias('mutatedSamples'),
+            # Adding disease name:
+            f.col('CANCER').alias('Cancer_type_acronym'),
+            f.col('CANCER_NAME').alias('diseaseFromSource'),
+            # Adding score:
+            f.col('QVALUE_COMBINATION').alias('resourceScore'),
+            parse_source(f.col('REFERENCE')).alias('literature'),
+        )
+        .join(
+            disease_mapping.select(
+                'Cancer_type_acronym',
+                f.trim(f.col('EFO_id')).alias('diseaseFromSourceMappedId'),
+            ),
+            on='Cancer_type_acronym',
+            how='left',
+        )
+        .drop('Cancer_type_acronym')
+    )
+
+
+def intogen(
+    source: dict[str, str],
+    destination: str,
+    properties: dict[str, str],
+) -> None:
+    """Intogen evidence generation step.
+
+    Args:
+        source (dict[str,str]): pts source configuration
+        destination (str): pts destination configuration
+        properties (dict[str,str]): pts cession configuration
+    """
+    # Initialise session:
+    session = Session(app_name='orphanet', properties=properties)
+
+    # Read input files:
+    genes = session.spark.read.csv(source['genes'], sep=r'\t', header=True, inferSchema=True)
+    cohorts = session.spark.read.csv(source['cohorts'], sep=r'\t', header=True, inferSchema=True).drop('SAMPLES')
+    disease_mapping = session.spark.read.csv(source['disease_mapping'], sep=r'\t', header=True)
+
+    # Generate intogen evidence:
+    generate_evidence(genes, cohorts, disease_mapping).write.parquet(destination)
```

Comment:
The write operation is missing right? This returns a dataframe that needs to be written?